music21/midi/translate.py

```diff
@@ -1966,13 +2011,15 @@ def streamHierarchyToMidiTracks(
     # TODO: may need to shift all time values to accommodate
     # Streams that do not start at same time
 
-    # store streams in uniform list
+    # store streams in uniform list: _prepareStreamForMidi() ensures there are substreams
     substreamList = []
-    if s.hasPartLikeStreams():
-        for obj in s.getElementsByClass('Stream'):
+    for obj in s.getElementsByClass('Stream'):
+        # _prepareStreamForMidi() supplies defaults for these
+        if obj.getElementsByClass(('MetronomeMark', 'TimeSignature')):
+            # Ensure conductor track is first
+            substreamList.insert(0, obj)
```

Comment:
As above, should be a flag that this is supporting conductor tracks OR we will need to expose `_prepareStreamForMidi()` and note that this is assumed to already have happened.  I prefer the former.

Reply:
Thanks for having a look! I'm just diving in first with the bit I'm having trouble following.

_prepareStreamForMidi() is called earlier in this method, and now _prepare...() will always strip meta events from the (deepcopy of the) inputted parts and collect them in a conductor stream.Part; so there is no way to opt out of exporting a conductor track (nor should there be IMO). Could you elaborate on what you had in mind here?

Reply:
I mostly just mean that the docs just say (if I remember) that it takes in a Stream, but it's not just any Stream, it's a Stream that already has had `._prepareStreamForMidi()` run on it, so since this is a public function, we probably need to expose `._prepareStreamForMidi()` as a public function and tell people to run that first and probably need to show running that first in the docs.  If the function can't run on a normal Stream, then that's fine, but neither of us want to answer questions on the mailing list that the function didn't work, because the asker didn't run `.prepareStreamForMidi()` first.

Reply:
Got it, then only the other place is an issue; this place can accept any stream and will always have _prepare run.